torch/distributed/checkpoint/_fsspec_filesystem.py

```diff
@@ -114,7 +115,7 @@ class _OverlappingCpuLoader(_TensorLoader):
     def __init__(
         self,
         resolve_fun: Callable,
-        stream: Union[None, io.RawIOBase, torch._C._CudaStreamBase] = None,
+        stream: Union[None, io.RawIOBase, torch.Stream] = None,
```

Comment:
The problem of using torch.Stream is, AFAICT, it won't typecheck `torch.cuda.Stream` as the later doesn't inherit from the former. We likely need to use a protocol instead.

Reply:
> The problem of using torch.Stream is, AFAICT, it won't typecheck `torch.cuda.Stream` as the later doesn't inherit from the former. We likely need to use a protocol instead.

ehah, in fact, the `torch.cuda.Stream` is inherited from `torch.Stream`, the relationship between two class is as follows:
![image](https://github.com/pytorch/pytorch/assets/22924371/f7ffab26-c5a8-4b3a-875e-c9d52f3facd2)

And I have tested it, 
```
>>> import torch
>>> issubclass(torch.cuda.Stream, torch.Stream)
True
>>> torch.__version__
'2.1.0a0+git1204463'
```

So maybe it is ok to use `torch.Stream`? @kumpera